### PR TITLE
additional features and functionality

### PR DESCRIPTION
**Pull request by SaraYaffe**

### CODE REVIEW

File: src/main/java/apod/ApodFrame.java [hunk 1]
```diff
@@ -75,41 +94,56 @@ public ApodFrame() {
...
+                        try {
+                            URI uri = new URI(videoUrl.toString());
+                            desktop.browse(uri);
+                        } catch (IOException | URISyntaxException ex) {
+                            ex.printStackTrace();
+                        }
+                    }
+                }
+            }
+        });
```
Comment (andrewoid) [inline src/main/java/apod/ApodFrame.java:142]: Doing this adds the MouseListener more than once when `setVideoUrl()` is called again.

Only add the mouse listener once.

File: src/main/java/apod/ApodFrame.java [hunk 2]
```diff
@@ -120,10 +154,26 @@ public void setPhotoTitle(String apodTitle) {
...
 
+    private void onSubmitClicked(ActionEvent e) {
+        presenter.downloadPhoto();
+
+    }
+
+    public void download(URL url) {
+        try {
+            File file = new File(//obv wont work for the user - fix
+                    "C:\\Users\\sarab\\IdeaProjects\\nasa_api\\saved_photos\\photo1.jpg");
```
Comment (andrewoid) [inline src/main/java/apod/ApodFrame.java:165]: Save it to the desktop

**Comment by andrewoid** [comment]
Do not add `saved_photos/photo1.jpg` to git

**Review by andrewoid — APPROVED** [review]

File: src/main/java/apod/ApodFrame.java [hunk 3]
```diff
@@ -120,10 +151,27 @@ public void setPhotoTitle(String apodTitle) {
...
+
+    public void download(URL url) {
+
+        try {
+            File desktop = new File(System.getProperty("user.home"), "/Desktop");
+            FileUtils.copyURLToFile(url, new File(desktop, "apod " + datePicker.getDate()));
+        } catch (IOException ex) {
+            ex.printStackTrace();
+        }
+    }
```
[the remark this answers is not in the document]
Comment (SaraYaffe) [reply src/main/java/apod/ApodFrame.java:168]: can datePicker in the view be public then? or would I just remove that reference from the new file name.

**Pull request merged**